Print deprecation warnings for extensions using $() on EventDispatchers

peterflynn opened the pull request:
Print deprecation warnings for extensions that are still using `$()` wrapper on Brackets EventDispatcher objects. The shim still works, so extensions have more time to migrate.

Many extensions have yet to migrate, so this creates quite a bit of console spam.  ~~That makes me a _little_ hesitant to land it late-ish in the 1.2 cycle -- maybe it would be better to land early in 1.3?  Open to either option, but I'm curious to hear others' opinions...~~ It now only prints one warning per caller, so the spam seems manageable -- I'd like to land this for 1.2 if it seems safe enough to others.

peterflynn committed:
Print deprecation warnings for extensions that are still using $() wrapper
on Brackets EventDispatcher objects. The shim still works, so extensions
have time to migrate.

MarcelGerber commented:
You may be able to introduce a new method in DeprecationWarning to only log a warning once per caller line.

peterflynn committed:
Only print deprecation warnings for $().on/off() on EventDispatchers once
per caller, like other such warnings. Move $() shim into brackets.js normal
startup code, since there's no longer any core code that could hit it (only
extensions).

peterflynn commented:
@MarcelGerber That was a good call!  DeprecationWarning has such a method already, but it's not reachable from main.js since it's so early in the startup sequence -- _but_ since we only need this shim for extension code now, I can move it into brackets.js where it's easy to call DeprecationWarning.  Changes pushed!

nethip commented:
Merged to master.
